### PR TITLE
vstart.sh: disable pg-autoscaling for crimson-osd

### PR DESCRIPTION
**Pull request by Matan-B**
Signed-off-by: Matan Breizman <mbreizma@redhat.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by ronen-fr — APPROVED** [review]

File: src/vstart.sh [hunk 1]
```diff
@@ -1314,6 +1315,11 @@ else
         debug ms = 1'
 fi
 
+if [ "$ceph_osd" == "crimson-osd" ]; then
```
Comment (rzarzynski) [inline src/vstart.sh:1318]: Might be worth leaving a comment explaining we don't support PG merge / split yet.

File: src/vstart.sh [hunk 2]
```diff
@@ -1314,6 +1315,11 @@ else
...
+    AUTOSCALER_OPTS='
+        osd_pool_default_pg_autoscale_mode = off'
+fi
+
```
Comment (rzarzynski) [inline src/vstart.sh:1322]: Not sure we need `src/` in the commit title. How about:
`vstart.sh: disable pg-autoscaling for crimson-osd`?

**Review by rzarzynski — APPROVED** [review]
LGTM, less-than-cosmetics only.

**Review by athanatos — APPROVED** [review]

**Comment by athanatos** [comment]
retest this please

**Comment by Matan-B** [comment]
jenkins retest this please

**Comment by athanatos** [comment]
retest this please

**Comment by rzarzynski** [comment]
jenkins retest this please

**Pull request merged**